ls1intum/Artemis
# `Development`: Remove public modifier from server test cases

### Checklist
#### General

### Motivation and Context
JUnit 4 forced test cases to have public visibility. With JUnit 5 this is no longer the case, in fact it is best practice to not use public as a visibility modifier for test cases. See https://junit.org/junit5/docs/current/user-guide/#writing-tests-classes-and-methods and https://rules.sonarsource.com/java/RSPEC-5786.


### Description
Search and Replace. Removed public from test cases and test classes.

The additional removed lines are because I removed an empty method [in the FeedbackSelectorTest](https://github.com/ls1intum/Artemis/pull/5324/files#diff-2126dee35cc06b3a4ce6658076badabc8e7c005b3e1c680f4f5b7ac274e2b65f).


### Steps for Testing
code review; check that the tests are still successful.

This PR adds a new PMD-Rule that checks against public test cases. So you can just open the Codacy-Report and check that there are no such style issues present.

### Review Progress

#### Code Review
